index.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+# Override the github.* context env vars with the ones that we got from the user
+# that may (or may not be) different from the original $GITHUB_* env vars.
+export GITHUB_TOKEN="$INPUT_TOKEN"
+export GITHUB_SERVER_URL="$INPUT_GITHUB_SERVER_URL"
+export GITHUB_REPOSITORY="$INPUT_REPOSITORY"
+
+# This is the default host that gh uses for clones and commands without a repo
+# context (a .git folder). We use Bash string magic to get the github.com part
+# from a full origin (no pathname) like https://github.com -> github.com
+export GH_HOST="${GITHUB_SERVER_URL#*//}"
+
+echo '🟪 Configuring local Git using gh auth setup-git...'
+gh auth setup-git
```

Comment:
@spenserblack are you familiar enough with gh cli to know what this does and if its needed? 🤔

Reply:
AFAIK it sets up `git` to use `gh` as the credential manager for the "github.com" domain. If you don't call any git commands that require authentication (`git pull`, `git push`), then I don't believe it's necessary. I've never used `gh auth setup-git` in a workflow, though, so I can't say for certain what the effect will be.